src/Common/QueryProfiler.cpp

```diff
@@ -31,6 +31,8 @@ namespace
     /// And so to overcome this flag is exists,
     /// to ignore delivered signals after timer_delete().
     thread_local bool signal_handler_disarmed = true;
+#else
+    thread_local bool signal_handler_disarmed = false;
```

Comment:
Let's not create unused variable for osx, but instead, fix it in the `QueryProfilerBase<ProfilerImpl>::tryCleanup()`.

And speaking about `QueryProfiler`, AFAICS `timer_t`/`timer_create`/`timer_settime`/`timer_delete` is not exists in osx, and it should complain on it too, no?

Reply:
Ok, let's simply move `signal_handler_disarmed` out of the `#if defined(OS_LINUX)`

Reply:
Here it is - https://github.com/ClickHouse/ClickHouse/pull/32219/commits/37ae9f943013448b42b57d0527adcbc40bcb52ac

P.S. sorry I cannot push into your topic branch, hence in a separate PR.

Reply:
that's ok ,just slove the problem